app/mqtt_reader.py

```diff
@@ -84,8 +93,6 @@ def inner(client, _userdata, _flags, rc):
                 for topic in self._topics:
                     logging.info("Subscribing to topic: %s", topic)
                     client.subscribe(topic)
-            else:
-                logging.error("Connection failed")
```

Comment:
?? why is this?
we aren't gonna be getting connection failures anymore?

Reply:
This came up in the last one - we're in the `on_connect` callback here, so the only way we would reach this line is if the connect method executes without throwing an `OSError` exception, but then calls the callback with a non-success flag - i.e. if it seems to succeed but then fails. It required a really fiddly test to get coverage for that scenario and I felt like it wasn't worth it. Now that I write that all out though, I guess it is possible that it fails in some other way 🙄 I'll put it back haha

Reply:
ok I've reinstated the logging with a slightly more specific message, and the test